### PR TITLE
Feature 8 add graph conversion tests

### PR DESCRIPTION
**Pull request by energyGiver**
# Background
This PR addresses issue #8 by improving the testing and error handling related to model loading.

# Changes
- Added integration tests for three models:
    - The basic model with fixed input arguments
    - A model for testing GRU conversion in a subgraph to analyze audio signals and tabular data
    - A model with fewer parameters, including a transformer
- Added error propagation for UnableToReadModel to make the root cause clearer.
- Modified tests to ensure an error is returned when the model path is missing, instead of ending with Ok.

# ToDo
I tried converting various models using Netron, but failed to load many models. 
After this PR, aside from the next task, I wanna keep focusing on analyzing the reason and resolving the following two cases:

- Analyze load failures for models split into .onnx and .onnx.data files.
- ~~Analyze load failures for models with components like GroupQueryAttention, TopK, etc~~ => Unimplemented operation in ONNX

### CODE REVIEW

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -87,6 +62,23 @@ fn test_model_graph_traversal() {
         output: Visibility::Public,
     };
 
+    let model_path = build_model_path("models/smol-llm.onnx");
```
Comment (chris-chris) [inline tests/integration_tests.rs:65]: LLM model
I think we have to implement most of the Operators to make the proof system work.

Comment (chris-chris) [reply tests/integration_tests.rs:65]: and ONNX file larger than 200MB is not suitable for the github repo file.
We should limit the binary file size or we should upload it to S3 bucket.

**Review by chris-chris — APPROVED** [review]
Good job.
Good practice.

File: tests/integration_tests.rs (hunk 1 above)
Comment (energyGiver) [reply tests/integration_tests.rs:65]: > I think we have to implement most of the Operators to make the proof system work.

Great, if we can create operations for transformers, I think we can port most models.

> We should limit the binary file size or we should upload it to S3 bucket.

I'll use the DVC to improve it during refactoring, thanks!

**Comment by chris-chris** [comment]
@energyGiver please fix the conflicts

**Pull request merged**